video/out/wayland_common.c

```diff
@@ -2830,8 +2847,13 @@ static void set_color_management(struct vo_wayland_state *wl)
     wp_image_description_creator_params_v1_set_primaries_named(image_creator_params, primaries);
     wp_image_description_creator_params_v1_set_tf_named(image_creator_params, transfer);
 
+    if (pl_color_transfer_is_hdr(color.transfer))
+        pl_color_space_infer(&wl->target_params.color);
```

Comment:
How about we sanitize also SDR metadata? Would this work? Stripe luminance data from SDR, because we likely don't want to set it.

```suggestion
    pl_color_space_infer(&wl->target_params.color);
    if (!pl_color_transfer_is_hdr(color.transfer)) {
        ...->min_luma = 0;
        ...->max_luma = 0;
        ...->max_cll = 0;
        ...->max_fall = 0;
    }
```

Reply:
On other APIs we don't do that. So for consistency we can skip that here too. I wanted to set that but was rejected on libplacebo side.

Either way for SDR we should have full coverage by setting named primaries. We set those?

Reply:
Named primaries and named transfers are always set. Right on master branch, setting metadata for SDR isn't skipped so if there are values in there, they would be set. I would assume compositors just ignore metadata for sdr anyway but never checked. With this PR, we would skip setting metadata on SDR completely.

It's technically possible to set primaries and the transfer function directly via values, but that's not implemented atm and would be another couple of functions completely. I'm not sure if compositors support that.

Reply:
Added a commit that actually does that (mistakenly thought the previous one did).